core/services/head_tracker.go

```diff
@@ -252,7 +243,11 @@ func (ht *HeadTracker) receiveHeaders() error {
 			if !open {
 				return errors.New("HeadTracker headers prematurely closed")
 			}
-			if err := ht.handleNewHead(blockHeader); err != nil {
+			if blockHeader == nil {
+				logger.Warn("got nil block header")
```

Comment:
This should probably be an error... I can't imagine this happening in a non-pathological case.